src/contracts/core/DelegationManager.sol

```diff
@@ -842,14 +897,49 @@ contract DelegationManager is
         return _operatorDetails[operator].stakerOptOutWindowBlocks;
     }
 
+    /**
+     * @notice Returns the migratable shares for an operator across given strategies and overwrite them to 0
+     * @param operator the operator to get migratable shares for
+     * @param allocator the allocator to get migratable shares to migrate to
+     * @param strategies the strategies to get migratable shares for
+     * @dev if any of the shares are 0, it will revert
+     */
+    function getMigratableOperatorShares(address operator, address allocator, IStrategy[] calldata strategies) external returns (uint256[] memory) {
+        // TODO: onlyAllocatorRegistry
+        require(_handoffs[operator].allocator == allocator, "DelegationManager.getMigratableOperatorShares: allocator mismatch");
+        require(isCompletableHandoff(operator), "DelegationManager.getMigratableOperatorShares: handoff not completable");
+        uint256[] memory shares = _getOperatorShares(operator, strategies);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            // overwrite to 0 because they're migrated
+            require(shares[i] != 0, "DelegationManager.getMigratableOperatorShares: operator has active shares");
```

Comment:
this require seems unnecessary? I think this just introduces a race condition in multiple calls to migrate operator shares with no clear benefit?